File: src/main/java/com/hileco/cortex/instructions/bits/BITWISE_XOR.java
package com.hileco.cortex.instructions.bits;

import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
public class BITWISE_XOR extends BitInstruction {
    public byte innerExecute(byte left, byte right) {
        byte result = left;
        result ^= right;
        return result;
    }
}

File: src/main/java/com/hileco/cortex/instructions/bits/BITWISE_XOR.kt
package com.hileco.cortex.instructions.bits

import com.hileco.cortex.constraints.expressions.Expression
import kotlin.experimental.xor

class BITWISE_XOR : BitInstruction() {
    override fun innerExecute(left: Byte, right: Byte): Byte {
        return left xor right
    }

    override fun innerExecute(left: Expression, right: Expression): Expression {
        return Expression.BitwiseAnd(left, right)
    }
}